### ISSUE TITLE
no-unreachable does not detect top-level unreachable statements

### ISSUE THREAD

**Issue by michaelficarra** [eslint/eslint#1138]
``` js
a()
throw a;
b();
```

`b()` is unreachable.
